.github/workflows/build.yml

```diff
@@ -259,3 +259,25 @@ jobs:
       - run: |
           echo "Some of the required build and test workflows have failed!"
           exit 1
+
+  notify-completed-successfully-failures:
+    if: ${{ always() && needs.completed-successfully.result == 'failure' && (github.ref_name == 'main' || startsWith(github.ref_name, 'release/')) }}
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: read
+    strategy:
+      fail-fast: false
+    needs:
+      - completed-successfully
+    steps:
+      - name: send-notification
+        uses: slackapi/slack-github-action@007b2c3c751a190b6f0f040e47ed024deaa72844 # v1.23.0
+        # We intentionally aren't using the following here since it's from an internal repo
+        # uses: hashicorp/cloud-gha-slack-notifier@730a033037b8e603adf99ebd3085f0fdfe75e2f4 #v1
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+        with:
+          channel-id: "C05AABYEA9Y" # sent to #feed-vault-ci-official
+          payload: |
+            {"text":"OSS build failures on ${{ github.ref_name }}","blocks":[{"type":"header","text":{"type":"plain_text","text":":flashing-light: OSS build failures :flashing-light:","emoji":true}},{"type":"divider"},{"type":"section","text":{"type":"mrkdwn","text":"build(s) failed on ${{ github.ref_name }}"},"accessory":{"type":"button","text":{"type":"plain_text","text":"View Failing Workflow","emoji":true},"url":"${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}"}}]}
```

Comment:
when testing this payload on [the slack block kit builder](https://app.slack.com/block-kit-builder/T024UT03C#%7B%22text%22:%22OSS%20build%20failures%20on%20main%22,%22blocks%22:%5B%7B%22type%22:%22header%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22:flashing-light:%20OSS%20build%20failures%20:flashing-light:%22,%22emoji%22:true%7D%7D,%7B%22type%22:%22divider%22%7D,%7B%22type%22:%22section%22,%22text%22:%7B%22type%22:%22mrkdwn%22,%22text%22:%22build(s)%20failed%20on%20main%22%7D,%22accessory%22:%7B%22type%22:%22button%22,%22text%22:%7B%22type%22:%22plain_text%22,%22text%22:%22View%20Failing%20Workflow%22,%22emoji%22:true%7D,%22url%22:%22https://github.com/hashicorp/vault/actions/runs/1234%22%7D%7D%5D%7D), I see that it errors with "invalid additional property 'text'". Is that a problem?

Reply:
Huh, weird. `text` doesn't actually change the message, but essentially provides info for screen-readers etc. When I was first adding these notifications, I got a warning if I didn't include it. I think we're good!

Reply:
I think the Slack GHA must use a different format to the builder you linked

Reply:
I think maybe this is using 'approach 2' as listed on this README: https://github.com/slackapi/slack-github-action